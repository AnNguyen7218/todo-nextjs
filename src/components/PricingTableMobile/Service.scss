@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.container {
  display: flex;
  align-items: center;
  margin-bottom: gutter(8);
  width: 100%;

  @include break(md) {
    width: 50%;
  }
}

.icon {
  width: 1.6rem;
  height: 1.6rem;
  color: $green;
  margin-right: gutter(10);
}
