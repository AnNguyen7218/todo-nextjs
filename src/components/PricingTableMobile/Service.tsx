import React from 'react';
import { ServiceRowOrder, PlanService } from 'types/pricingTable';
import Checked from 'assets/icons/ic-checked.svg';
import { ICON } from 'constants/pricingTable';
import css from './Service.scss';

type Props = {
  service: ServiceRowOrder;
  planServices: PlanService;
};

const Service: React.FC<Props> = ({ service, planServices }) => {
  const { serviceKey, name, type } = service;
  const value = planServices[serviceKey];

  if (!value) {
    return null;
  }

  if (type === ICON) {
    return (
      <div className={css['container']}>
        <Checked className={css['icon']} />
        <p>{name}</p>
      </div>
    );
  }

  if (typeof value === 'string' && !isNaN(Number(value))) {
    return (
      <div className={css['container']}>
        <Checked className={css['icon']} />
        <p>
          {value} {name}
        </p>
      </div>
    );
  }
  return (
    <div className={css['container']}>
      <Checked className={css['icon']} />
      <p>{value}</p>
    </div>
  );
};

export default Service;
