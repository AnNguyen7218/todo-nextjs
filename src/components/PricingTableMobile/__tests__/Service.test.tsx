import React from 'react';
import { render, cleanup } from '@testing-library/react';
import {
  STARTUP,
  PLAN_SERVICES,
  SERVICE_LISTING,
  SERVICES,
  VERIFIED_PROFILE,
  RECEIVE_RESPOND_INQUIRIES,
  CUSTOMER_SERVICE,
  PERSONALISED_CUSTOMER_SERVICE,
} from 'constants/pricingTable';
import Service from '../Service';

afterEach(cleanup);

describe('Service Component', () => {
  test('it should match snapshot', () => {
    const { container } = render(
      <>
        <Service
          service={SERVICES[SERVICE_LISTING]}
          planServices={PLAN_SERVICES[STARTUP]}
        />
        <Service
          service={SERVICES[VERIFIED_PROFILE]}
          planServices={PLAN_SERVICES[STARTUP]}
        />
        <Service
          service={SERVICES[RECEIVE_RESPOND_INQUIRIES]}
          planServices={PLAN_SERVICES[STARTUP]}
        />
        <Service
          service={SERVICES[CUSTOMER_SERVICE]}
          planServices={PLAN_SERVICES[STARTUP]}
        />
        <Service
          service={SERVICES[PERSONALISED_CUSTOMER_SERVICE]}
          planServices={PLAN_SERVICES[STARTUP]}
        />
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
