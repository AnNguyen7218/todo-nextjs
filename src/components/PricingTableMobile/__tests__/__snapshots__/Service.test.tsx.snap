// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Component it should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <ic-checked.svg
      classname="icon"
    />
    <p>
      1
       
      Service Listing
    </p>
  </div>
  <div
    class="container"
  >
    <ic-checked.svg
      classname="icon"
    />
    <p>
      Receive & respond to inquiries
    </p>
  </div>
  <div
    class="container"
  >
    <ic-checked.svg
      classname="icon"
    />
    <p>
      Email support
    </p>
  </div>
</div>
`;
